.github/workflows/CI.yml

```diff
@@ -1,5 +1,10 @@
-name: Tests
-on: [push, pull_request]
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```

Comment:
I'd add as well:

```suggestion
  pull_request:
  # Allow running manually the CI workflow.
  workflow_dispatch:

# This allows a subsequently queued workflow run to interrupt previous runs.
concurrency:
  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
  cancel-in-progress: true
```

found recently in gh help and applying to most of my projects!